### PR TITLE
Set xunit env var to not print output for passing tests

### LINKED ISSUE #103445: Tests are spamming ITestOutputHelper logs for passing tests
Tests that use `ITestOutputHelper` are now printing the output even when the test is passing.
For example, if you run `dotnet build .\src\libraries\System.Net.Http\tests\FunctionalTests\ /t:Test`, the console will print 5000 lines of text even if everything's passing.

It looks like it started with #103279

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/runtime/issues/103445

### CODE REVIEW

File: eng/testing/xunit/xunit.console.targets [hunk 1]
```diff
@@ -85,13 +92,6 @@
...
             Visible="false" />
     </ItemGroup>
-
-    <!-- Workaround for https://github.com/xunit/xunit/issues/1651 -->
-    <ItemGroup Condition="'$(ArchiveTests)' != 'true'">
-      <None Remove="$(Pkgxunit_runner_visualstudio)\build\net452\xunit.runner.utility.net452.dll" />
-      <None Remove="$(Pkgxunit_runner_visualstudio)\build\net452\xunit.runner.reporters.net452.dll" />
-      <None Remove="$(Pkgxunit_runner_visualstudio)\build\netcoreapp2.1\xunit.runner.utility.netcoreapp10.dll" />
-      <None Remove="$(Pkgxunit_runner_visualstudio)\build\netcoreapp2.1\xunit.runner.reporters.netcoreapp10.dll" />
-    </ItemGroup>
```
Comment (ViktorHofer) [inline eng/testing/xunit/xunit.console.targets:95]: Unrelated change: This isn't necessary anymore as the underlying bug was fixed in xunit.

**Review by jkoritzinsky — APPROVED** [review]

**Review by ManickaP — APPROVED** [review]
Miha's OOF, I tested it, LGTM, thank you!

**Comment by carlossanlop** [comment]
How will the output look from now on if all tests succeed?
When there's a combination of failed and succeeded tests, will it look confusing for folks unaware of this change?
Do you think this change is noticeable enough to announce it to the team?

**Comment by ViktorHofer** [comment]
To answer all of your questions, please see the linked issue. This is restoring the previous behavior, before we updated to xunit 2.8.1. The new xunit behavior (starting with 2.8.1) is the one that team member noticed and complained about.
